stable/appmesh-jaeger/templates/deployment.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "appmesh-jaeger.fullname" . }}
+  labels:
+{{ include "appmesh-jaeger.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "appmesh-jaeger.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "appmesh-jaeger.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "14269"
+    spec:
+      serviceAccountName: {{ include "appmesh-jaeger.serviceAccountName" . }}
+      volumes:
+      - name: data
+        {{- if .Values.persistentVolumeClaim.claimName }}
+        persistentVolumeClaim:
+          claimName: {{ .Values.persistentVolumeClaim.claimName }}
+        {{- else }}
+        emptyDir: {}
+        {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
```

Comment:
I found this reference in the jaeger docs for a description of what all 9 of these ports do: 

Port | Protocol | Component | Function
-- | -- | -- | --
5775 | UDP | agent | accept zipkin.thrift over compact thrift protocol (deprecated, used by legacy clients only)
6831 | UDP | agent | accept jaeger.thrift over compact thrift protocol
6832 | UDP | agent | accept jaeger.thrift over binary thrift protocol
5778 | HTTP | agent | serve configs
16686 | HTTP | query | serve frontend
14268 | HTTP | collector | accept jaeger.thrift directly from clients
9411 | HTTP | collector | Zipkin compatible endpoint (optional)

Can we add a small descriptive comment to each one?  Also, it looks like they don't line up completely, what are all the other ports in the 14k range?

Reply:
I've documented the ports, the 14k ports are for health checking see https://www.jaegertracing.io/docs/1.14/monitoring/